Firebase/Core/FIRApp.m

```diff
@@ -107,15 +108,22 @@ + (void)configure {
     // since the app isn't configured at all, so only rely on the Info.plist value.
     NSNumber *collectionEnabledPlistValue = [[self class] readDataCollectionSwitchFromPlist];
     if (collectionEnabledPlistValue == nil || [collectionEnabledPlistValue boolValue]) {
-      [[NSNotificationCenter defaultCenter]
-          postNotificationName:kFIRAppDiagnosticsNotification
-                        object:nil
-                      userInfo:@{
-                        kFIRAppDiagnosticsConfigurationTypeKey : @(FIRConfigTypeCore),
-                        kFIRAppDiagnosticsErrorKey : [FIRApp errorForMissingOptions]
-                      }];
+      if (NSClassFromString(@"FIRDiagnostics")) {
+        FIRDiagnosticsData *diagnosticsData = [[FIRDiagnosticsData alloc] init];
+        [diagnosticsData insertValueIfNotNil:collectionEnabledPlistValue
+                                      forKey:kFIRCDIsDataCollectionDefaultEnabledKey];
+        [diagnosticsData insertValueIfNotNil:[FIRApp firebaseUserAgent]
+                                      forKey:kFIRCDFirebaseUserAgentKey];
+        [diagnosticsData insertValueIfNotNil:@(FIRConfigTypeCore)
```

Comment:
I wonder, how much of this logic we may move to e.g. a separate class? I see some fields are set for each case as well as the notification is posted. I think, having methods like `-[FIRDiagnosticsInterop postDiagnosticDataNotificationWith<Something>:]` can improve readability a lot.

Reply:
Agreed, but no other interop lib has logic in it. Is that something that's allowed? @ryanwilson